File: LeetCode/LeetCode/Problems/19/Problem1921.cs
using System;
using System.Linq;

public class Problem1921
{
    public int EliminateMaximum(int[] dist, int[] speed)
    {
        var minutesToReach = dist.Zip(speed)
            .Select(x =>
            {
                var dist = x.First;
                var speed = x.Second;
                return dist / speed + (dist % speed > 0 ? 1 : 0);
            })
            .ToArray();

        Array.Sort(minutesToReach);

        var n = minutesToReach.Length;

        for (int i = 0; i < n; i++)
        {
            if (i >= minutesToReach[i])
                return i;
        }

        return n;
    }
}

File: leetcode_rust/src/problems/problem_1921.rs
pub fn eliminate_maximum(mut dist: Vec<i32>, speed: Vec<i32>) -> i32 {
    let n = dist.len() as i32;
    for i in 0..(n as usize) {
        let dist_c = dist[i];
        let speed_c = speed[i];
        dist[i] = dist_c / speed_c + if dist_c % speed_c > 0 { 1 } else { 0 };
    }

    dist.sort();

    for i in 0..n {
        if i >= dist[i as usize] {
            return i;
        }
    }

    n
}
